tests/unit/cli/vyper_json/test_compile_json.py

```diff
@@ -319,3 +320,37 @@ def test_compile_json_with_abi_top(make_input_bundle):
     """
     input_bundle = make_input_bundle({"stream.json": stream, "code.vy": code})
     vyper.compiler.compile_code(code, input_bundle=input_bundle)
+
+
+def test_compile_json_with_experimental_codegen():
+    code = {
+        "language": "Vyper",
+        "sources": {"foo.vy": {"content": "@external\ndef foo() -> bool:\n    return True"}},
+        "settings": {
+            "evmVersion": "cancun",
+            "optimize": "gas",
+            "venom": True,
+            "search_paths": [],
+            "outputSelection": {"*": ["ast"]},
+        },
+    }
+
+    settings = get_settings(code)
+    assert settings.experimental_codegen is True
+
+
+def test_compile_json_with_both_venom_aliases():
+    code = {
+        "language": "Vyper",
+        "sources": {"foo.vy": {"content": ""}},
+        "settings": {
+            "evmVersion": "cancun",
+            "optimize": "gas",
+            "experimentalCodegen": False,
+            "venom": False,
+            "search_paths": [],
+            "outputSelection": {"*": ["ast"]},
+        },
+    }
+    with pytest.raises(JSONError):
+        get_settings(code)
```

Comment:
maybe check the error message
```suggestion
    with pytest.raises(JSONError):
        get_settings(code)
    assert "both experimentalCodegen and venom cannot be set" in str(e.value)
```

Reply:
https://github.com/vyperlang/vyper/pull/4337/commits/d6bba9917f4e168c0005427a110e6d615cf732e4